src/simple-cache-client.ts

```diff
@@ -252,51 +262,55 @@ export class SimpleCacheClient {
    * Add a value to the beginning of a list.
    * @param {string} cacheName - Name of the cache with the list.
    * @param {string} listName - The list to push to.
-   * @param {string | Uint8Array} - The value to push.
-   * @param {CollectionTtl} [ttl] - How the ttl should be managed. Defaults to the client's TTL.
-   * @param {number} [truncateFrontToSize] - If the list exceeds this length, remove excess from the start of the list. Must be positive.
+   * @param {string | Uint8Array} value - The value to push.
+   * @param {FrontTruncatableCallOptions} [options]
```

Comment:
I'm typically all for DRY programming, but I'm wondering if reusing this type across multiple calls will be intuitive for users. I might think there was a typo in the intellisense if I was trying to autocomplete the params list for `listPopFront` and got prompted with `FrontTruncatableCallOptions`.

I'm asking this open ended wondering what the consensus is of the dev team. I think we either:

* Leave it as is and make sure we document it very well
* Create types for every call

Pro for leaving it as is is that we have fewer classes to manage. Con is that if we add something specific to one call but not the others, we have to introduce a breaking change because the signature would change.

Reply:
2 thoughts on this:

1. I think we actually could make it a backward compatible change because TS/JS interfaces are really flexible; as long as the new interface in the future is a superset of the old one then it shouldn't be a breaking change.
2. I think we could maybe get the best of both worlds by just doing a per-API type alias?  so like:

```typescript
type ListPushFrontOptions = FrontTruncateCallOptions
```

If we just added one line like that above each API then I think the type hints would hopefully give a less weird-sounding name but there's not really anything extra for us to maintain?

Reply:
Big fan of number 2!

Reply:
Truncatable is a word in a math context at least - [https://en.wikipedia.org/wiki/Truncatable_prime](url). It not being in the dictionary is unfortunate.
We would add a new sub-interface if we had a new method that needed an additional optional argument. We couldn't add a new optional argument to an existing signature without either making it available to other methods that use the same options object or changing which one it takes. Is the convenience of not technically breaking backwards compatibility when adding an optional argument worth the overhead of managing per-call options classes? @cprice404